src/inet/IPEndPointBasis.cpp

```diff
@@ -1292,20 +1294,28 @@ void IPEndPointBasis::GetPacketInfo(nw_connection_t aConnection, IPPacketInfo &
     aPacketInfo.DestPort    = nw_endpoint_get_port(dest_endpoint);
 }
 
-INET_ERROR IPEndPointBasis::GetEndPoint(nw_endpoint_t & aEndPoint, const IPAddress aAddress, uint16_t aPort)
+INET_ERROR IPEndPointBasis::GetEndPoint(nw_endpoint_t & aEndPoint, const IPAddressType aAddressType, const IPAddress & aAddress,
+                                        uint16_t aPort)
 {
     INET_ERROR res = INET_NO_ERROR;
-
     char addrStr[INET6_ADDRSTRLEN];
-    aAddress.ToString(addrStr, sizeof(addrStr));
-
     char portStr[INET_PORTSTRLEN];
-    snprintf(portStr, sizeof(portStr), "%u", aPort);
 
-    nw_endpoint_t endpoint = nw_endpoint_create_host(addrStr, portStr);
-    VerifyOrExit(endpoint != NULL, res = INET_ERROR_BAD_ARGS);
+    // Note: aAddress.ToString will return the IPv6 Any address if the address type is Any, but that's not what
```

Comment:
Ah, I now see why you'd done what you'd done before. This is an interesting side effect of IPAddress::Type() and kIPAddressType_Any.

Given that, despite the copy present before, I think I prefer the copy to the bare "0.0.0.0". What do you think?

Either way, whichever path you decide, the comment should stay.

Reply:
Argh ! Justin merged just before I push an updated version ! I will fix that in a followup I also do not like the bare string.

Reply:
Sorry, I saw the approval :(

Reply:
Did a followup in #1423